apps/sms/views/conversation/js/conversation.js

```diff
@@ -2026,7 +1886,18 @@ var ConversationView = {
         text: 'delete',
         className: 'danger'
       }
-    ).then(performDeletion.bind(this));
+    ).then(() => {
+      WaitingScreen.show();
+
+      var messageIdsToDelete = this.selectionHandler.selectedList.map(
+        (item) => +item
+      );
+
+      return this.deleteMessages(messageIdsToDelete).then(() => {
+        ConversationView.cancelEdit();
+        WaitingScreen.hide();
+      });
```

Comment:
same thing here: we can move the then in the outer promise chain; it makes sense also.